.github/workflows/preview-env-build-and-deploy.yml

```diff
@@ -0,0 +1,217 @@
+        with:
+          ref: refs/heads/${{ github.head_ref }}
+          fetch-depth: 0 # fetches all history for all branches and tags
+
+      - uses: ./.github/actions/setup-zeebe
+        with:
+          secret_vault_secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secret_vault_address: ${{ secrets.VAULT_ADDR }}
+          secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
+
+      - uses: ./.github/actions/build-zeebe
+        id: build-zeebe
+        with:
+          maven-extra-args: -T1C -PskipFrontendBuild
+
+      - name: Import Secrets
+        id: secrets
+        uses: hashicorp/vault-action@v3.0.0
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/data/github.com/organizations/camunda NEXUS_USR;
+            secret/data/github.com/organizations/camunda NEXUS_PSW;
+
+      - name: Set environment variables
+        env:
+          HEAD_REF: ${{ github.head_ref }}
+        run: |
+          GIT_COMMIT_HASH=$(git rev-parse $HEAD_REF)
+          echo "VERSION=pr-$GIT_COMMIT_HASH" >> $GITHUB_ENV
+
+      - name: Login to Harbor docker registry
+        uses: docker/login-action@v3
+        with:
+          registry: registry.camunda.cloud
+          username: ${{ steps.secrets.outputs.NEXUS_USR }}
+          password: ${{ steps.secrets.outputs.NEXUS_PSW }}
+
+      - uses: ./.github/actions/build-platform-docker
+        with:
+          repository: registry.camunda.cloud/team-camunda/zeebe
+          version: ${{ env.VERSION }}
+          platforms: ${{ env.DOCKER_PLATFORMS }}
+          push: true
+          distball: ${{ steps.build-zeebe.outputs.distball }}
+
+  build-optimize:
+    if: github.event.pull_request.state != 'closed' && (contains( github.event.label.name, 'deploy-preview') || contains( github.event.pull_request.labels.*.name, 'deploy-preview'))
+    name: Optimize Build
+    uses: ./.github/workflows/optimize-ci-build-reusable.yml
+    secrets: inherit
+    with:
+      branch: ${{ github.head_ref }}
+      pushDocker: true
+
+  build-operate:
+    if: github.event.pull_request.state != 'closed' && (contains( github.event.label.name, 'deploy-preview') || contains( github.event.pull_request.labels.*.name, 'deploy-preview'))
+    name: Operate Build
+    uses: ./.github/workflows/operate-ci-build-reusable.yml
+    secrets: inherit
+    with:
+      branch: ${{ github.head_ref }}
+      previewEnv: true
+
+  build-tasklist:
+    if: github.event.pull_request.state != 'closed' && (contains( github.event.label.name, 'deploy-preview') || contains( github.event.pull_request.labels.*.name, 'deploy-preview'))
+    name: Tasklist Build
+    uses: ./.github/workflows/tasklist-ci-build-reusable.yml
+    secrets: inherit
+    with:
+      branch: ${{ github.head_ref }}
+      previewEnv: true
+
+  deploy-preview:
+    if: |
+        always() &&
```

Comment:
yes thats true - I actually implemented it that way because the Optimize workflow was failing but I am under the impression that it got fixed after rebasing the last changes form main